browser/component_updater/brave_component_installer.cc

```diff
@@ -48,8 +48,7 @@ bool RewriteManifestFile(const base::FilePath& extension_root,
 
   base::FilePath manifest_path =
       extension_root.Append(FILE_PATH_LITERAL("manifest.json"));
-  int size = base::checked_cast<int>(manifest_json.size());
-  if (base::WriteFile(manifest_path, manifest_json.data(), size) != size) {
```

Comment:
seems like any `return_value != size` (inlcuding `-1`) was treated as error, so it's not a bug. but maybe I'm missing something?

Reply:
The snippet in the right uses a different function in the overload set that takes as argument a string piece, and returns a `bool`.

Reply:
ah! this one is(was) fine, but other two in this PR were not fine.